src/com/chocohead/clm/blocks/MotorBlock.java

```diff
@@ -125,67 +136,80 @@ public void onScheduledTick(BlockState state, World world, BlockPos pos, Random
 			if (node != null) {
 				NetworkState networkState = node.getNetwork().getState();
 				int momentum = networkState.getMomentum();
+				int rotationMultiplier = state.get(ROTATION).multiplier;
 
-				if (momentum < 15) {//Add on some momentum so long as it's not going to be faster than 50
-					networkState.setMomentum(momentum + nextUpdate);
+				if (momentum * rotationMultiplier < 15) {//Add on some momentum so long as it's not going to be faster than 50
```

Comment:
Won't this need to be compared absolutely to avoid anticlockwise rotation being unbounded? Or does it always work out that the signs of `momentum` and `rotationMultiplier` will be the same when it matters?

Reply:
Think of it this way: For clockwise rotation (multiplier = 1), the motor tries to keep the momentum at 15 by only adding to the momentum. For counter-clockwise rotation (multiplier = -1), the motor tries to keep the momentum at -15 by only subtracting from the momentum. For counter-clockwise rotation, the condition is true iff momentum > -15.

Reply:
I had a feeling that's how it would work, all works out quite nicely really